### PR TITLE
workflows/openshift-os allow custom branches

### PR DESCRIPTION
**Pull request by jbtrystram**
Allow specifying the source branch to open a PR from
as well as the target branch to compare against.
This is useful for backporting work

Also update the action as there were some deprecation warnings

### CODE REVIEW

File: .github/workflows/openshift-os.yml [hunk 1]
```diff
@@ -77,10 +79,10 @@ jobs:
           EOF
 
       - name: Open pull request
-        uses: peter-evans/create-pull-request@v4.2.3
+        uses: peter-evans/create-pull-request@v6.0.0
         with:
           token: ${{ secrets.COREOSBOT_RELENG_TOKEN }}
-          push-to-fork: coreosbot-releng/os
+          push-to-fork: openshift/os
```
Comment (jlebon) [inline .github/workflows/openshift-os.yml]: This looks incorrect.

File: .github/workflows/openshift-os.yml [hunk 2]
```diff
@@ -67,7 +69,7 @@ jobs:
               echo "No non-trivial changes; exiting"
               exit 0
           fi
-          git checkout $SOURCE_BRANCH
```
Comment (jlebon) [inline .github/workflows/openshift-os.yml:70]: Without this, there's nothing actually updating the submodule, right?

**Review by jlebon — COMMENTED** [review]
My 2c is to rework this to use a `choice` input instead. 3 inputs is a lot to get right (esp. the `branch-name` one; forgetting to update that one would stomp on an open PR opened against a completely different branch -- though maybe the `create-pull-request` action would error out on that).

File: .github/workflows/openshift-os.yml (hunk 2 above)
Comment (jbtrystram) [reply .github/workflows/openshift-os.yml:70]: yes, that was what I got confused about yesterday !

**Comment by jbtrystram** [comment]
Ok, here is a working version of the workflow :) 

Successful run (on my fork to avoid noise on openshift/os ) : https://github.com/jbtrystram/fedora-coreos-config/actions/runs/8081693445/job/22080845871
Created PR : (here backporting 4.15 release) -> https://github.com/jbtrystram/openshift-os/pull/1

File: .github/workflows/openshift-os.yml [hunk 3]
```diff
@@ -6,22 +6,23 @@ on:
...
-        description: Target branch to open PR against
-        required: true
-        default: master
-
+        type: choice
+        options:
+          - testing-devel
+          - branched
+          - rawhide
+          - next-devel
```
Comment (jlebon) [inline .github/workflows/openshift-os.yml]: We can drop these. We never normally need to sync those.

File: .github/workflows/openshift-os.yml [hunk 4]
```diff
@@ -6,22 +6,23 @@ on:
...
+          - rhcos-4.10
+          - rhcos-4.11
+          - rhcos-4.12
+          - rhcos-4.13
+          - rhcos-4.14
+          - rhcos-4.15
```
Comment (jlebon) [inline .github/workflows/openshift-os.yml]: Is the order here the same order the choices are laid out when you trigger the action? If so, should we order it the other way instead? (I.e. 4.15 first, then down to 4.10.)

File: .github/workflows/openshift-os.yml [hunk 5]
```diff
@@ -30,35 +31,54 @@ jobs:
...
+                echo "BRANCH_NAME=fcc-sync" >> $GITHUB_ENV
+              ;;
+              rhcos-*)
+              # split the string around the -
+                array=(${SOURCE_BRANCH//-/ })
+                OCP_VERSION=${array[1]}
+                echo "TARGET_BRANCH=release-${OCP_VERSION}" >> $GITHUB_ENV
+                echo "BRANCH_NAME=fcc-sync-${SOURCE_BRANCH}" >> $GITHUB_ENV
+                ;;
+              *)
```
Comment (jlebon) [inline .github/workflows/openshift-os.yml:48]: We should just error out in this path since it shouldn't be reachable.

**Review by jlebon — COMMENTED** [review]
This looks great!

File: .github/workflows/openshift-os.yml (hunk 4 above)
Comment (jbtrystram) [reply .github/workflows/openshift-os.yml]: yes. Fixed.

File: .github/workflows/openshift-os.yml (hunk 5 above)
Comment (jbtrystram) [reply .github/workflows/openshift-os.yml:48]: I am not sure if the default values are set when running on the `schedule` event. the syntax does not make it look like so, so I'd keep that default case there IMO

Comment (jbtrystram) [reply .github/workflows/openshift-os.yml:48]: yep, further testing confirm this: the `on_schedule` event do not initialize the variables, so we need this block :)

Comment (jbtrystram) [reply .github/workflows/openshift-os.yml:48]: I pushed an update enforcing `testing-devel`, `master` and `fcc-sync` in that default case ( to keep consistency the previous action behaviour for the daily sync)

**Review by jlebon — APPROVED** [review]

**Pull request merged**